File: Algorithms/1351.Count-Negative-Numbers-in-a-Sorted-Matrix/count_negative_numbers_in_a_sorted_matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        m = len(grid)
        n = len(grid[0])
        res = 0
        j = n - 1
        for i in range(m):
            while (j >= 0 and grid[i][j] < 0):
                j -= 1 
            res += n - 1 - j
        return res

File: Algorithms/1351.Count-Negative-Numbers-in-a-Sorted-Matrix/count_negative_numbers_in_a_sorted_matrix.go
func countNegatives(grid [][]int) int {
    n := len(grid[0])
    res := 0
    for _, ele := range grid {
        for j := 0; j < n; j++ {
            if ele[j] < 0 {
                res = res + n - j
                j = n
            }
        }
    }
    return res
}
